### ISSUE TITLE
Rust: fix upstream sled documentation / code to indicate the need for flushes

### ISSUE THREAD

**Issue by brson** [pingcap/talent-plan#115]
Per @sticnarf sled's api requires explicit frequent flushes that aren't sufficiently described in its documentation. Fix that.

**Comment by sticnarf** [comment]
@brson https://github.com/spacejam/sled/issues/654
sled's reply on this

**Comment by brson** [comment]
Do the sled docs explicitly explain that data needs to be flushed in order for it to be written to disk? That's probably all we need.

**Comment by sticnarf** [comment]
No. Seems that they think OS pagecache can help data survive in most cases.

**Comment by brson** [comment]
I understand the author's position that it's not the engine's responsibility to do the flushing - that's more of a database policy thing. But what I am concerned about is that their docs don't sufficiently explain that sled _doesn't_ automatically flush, and that the user needs to implement _some_ flushing policy.

At this point though, although their docs might need to be improved, we probably should just mention it in our sled section.

**Comment by brson** [comment]
Of course, since we have no content at all about data integrity, it hardly _actually_ matters whether the user flushes anything since no tests will break, though I guess if I'm scoring their work I would hope to see flushes in the right places.

**Comment by aos** [comment]
I ran into this issue when my `sled` tests were failing here: https://github.com/pingcap/talent-plan/blob/a5dccffb09d8d996d9b20a45061cde6604cf221a/courses/rust/projects/project-3/tests/cli.rs#L297-L324
Since the server is "rebooted" pretty quickly, it's possible that dirty writes aren't flushed in time and those two tests will fail with sled without an explicit flush call. I guess these tests are inadvertently also checking for appropriate flushing :-)
